src/sentry/monitors/migrations/0011_backfill_monitor_environment_is_muted.py

```diff
@@ -0,0 +1,73 @@
+# Generated by Django 5.2.1 on 2025-11-13 19:27
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+
+
+def backfill_monitor_environment_is_muted(apps, schema_editor):
+    """
+    Backfill MonitorEnvironment.is_muted from Monitor.is_muted for all muted monitors.
+
+    After this migration:
+    - is_muted=True monitors will have ALL environments muted
+    - is_muted=False monitors will have environments unchanged (some, none, or all unmuted)
+
+    This gets us into the correct state for the dual-write implementation where:
+    - Monitor is muted if and only if ALL environments are muted
+    - Monitor is unmuted if ANY environment is unmuted
+    """
+    Monitor = apps.get_model("monitors", "Monitor")
+    MonitorEnvironment = apps.get_model("monitors", "MonitorEnvironment")
+
+    # Use range query to iterate over all muted monitors
+    # Only need to update environments for muted monitors since unmuted is already the default
+    batch_size = 1000
+    last_id = 0
+
+    while True:
+        muted_monitors = list(
+            Monitor.objects.filter(
+                id__gt=last_id,
+                is_muted=True,
+            ).order_by(
+                "id"
+            )[:batch_size]
+        )
```

Comment:
You may as well just use `RangeQuerySetWrapper` and filter `is_muted` in memory. There's no index on `id, is_muted` so writing this manually won't help at all.